app/src/main/java/com/mcso/ap/chromanews/ui/newsfeed/BusinessFragment.kt

```diff
@@ -56,6 +57,23 @@ class BusinessFragment: Fragment() {
         viewModel.fetchDone.observe(viewLifecycleOwner) {
             binding.swipeRefreshLayout.isRefreshing = false
         }
+
+        viewModel.observeSearchPostLiveData().observe(viewLifecycleOwner,
+            Observer { filterList ->
+                adapter.submitList(filterList)
+                adapter.notifyDataSetChanged()
+            }
+        )
+
+        // sentiment analyzer
+        viewModel.observeSentimentScore().observe(viewLifecycleOwner) { sentimentData ->
```

Comment:
adding sentiment analyzer to each fragment will cause the calls to happen thrice... for current, previous and next tabs. I have already moved this under ReadNews.kt. Please remove this observer from fragments